src/Mappers/BaseMapper.php

```diff
@@ -62,17 +64,158 @@ protected static function getValueFromTag(\DOMDocument $document, string $tag):
         return $element->textContent;
     }
 
-    protected static function getField(\DOMElement $element, string $fieldTagName): ?string
+    protected static function getField(\DOMElement $element, string $fieldTagName, $object = null): ?string
     {
         $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
         if (!isset($fieldElement)) {
             return null;
         }
 
+        if (isset($object)) {
+            self::checkForMessage($object, $fieldElement);
+        }
+
         if ($fieldElement->textContent === "") {
             return null;
         }
 
         return $fieldElement->textContent;
     }
+    
+    protected static function checkForMessage($object, \DOMElement $element): void
```

Comment:
The object should have a typehint or implement an interface, e.g. `HasMessage`.

Reply:
I'm not really up to date on using interfaces and typehinting for multiple classes. Can you give me some help on this? Or is this something that we can address at a later stage?

Reply:
Well, this method calls `addMessage()` on the object, so it cannot take _any_ object right. It has to take an object that supports messages and the required method call. 

So ideally, these objects would have an interface `HasMessage` that defines the method `addMessage(Message $message): void` so that you can use this typehint in the method parameter list. 

You can read a bit about interfaces here: https://www.zentut.com/php-tutorial/php-interface/ Please ignore the crappy naming in the article, interfaces should have names like `SupportFoo`, `CanBar` or even `IsBazable` to indicate the class supports some _functionality_.

Reply:
Done and believe it works as intended now